### DISCUSSION TITLE
Does WarpX require GPU?

### DISCUSSION THREAD

**Discussion by erny123** [BLAST-WarpX/warpx#4192] [Q&A] [answered]
Wondering if WarpX requires a GPU? If so, are there any particular instructions that need to be followed during installation to disable GPUS usage?
Would love to try some of the mesh refinement and benchmark between a gpu enabled and disabled WarpX. 

Sorry, I couldn't find the answer anywhere.

**Comment by ax3l** [comment, marked as answer]
Hi @erny123,

Great question! No, WarpX does not require a GPU to run. WarpX can in fact be [run on either CPUs](https://warpx.readthedocs.io/en/latest/index.html#warpx), or three kind of different GPU vendors (Nvidia, AMD, Intel).

To run on CPUs, just change during compilation the flag [WarpX_COMPUTE](https://warpx.readthedocs.io/en/latest/install/cmake.html#build-options) to read `OMP` (OpenMP, our default to use multiple CPU cores) or `NOACC` (fully CPU serial, not really recommended unless OpenMP causes issues).
```bash
rm -rf build
cmake -S . -B build -DWarpX_COMPUTE=OMP
cmake --build build -j 8
```

We also provide pre-compiled CPU packages, e.g., via [conda](https://warpx.readthedocs.io/en/latest/install/users.html#using-the-conda-package).

So yes, you can run WarpX on CPU systems such as laptops, desktops or CPU clusters.

Let us know if you have any further questions! :)
